src/Riok.Mapperly/Descriptors/WellKnownTypes.cs

```diff
@@ -1,12 +1,17 @@
+using System.Collections.Immutable;
 using Microsoft.CodeAnalysis;
 using Riok.Mapperly.Helpers;
+using Riok.Mapperly.Symbols;
 
 namespace Riok.Mapperly.Descriptors;
 
 public class WellKnownTypes
 {
     private readonly Compilation _compilation;
     private readonly Dictionary<string, INamedTypeSymbol?> _cachedTypes = new();
+    private readonly Dictionary<ISymbol, ImmutableArray<AttributeData>> _cachedAttributes = new(SymbolEqualityComparer.Default);
+    private readonly Dictionary<ITypeSymbol, ISymbol[]> _allMembers = new(SymbolEqualityComparer.Default);
+    private readonly Dictionary<ITypeSymbol, IMappableMember[]> _allAccessibleMembers = new(SymbolEqualityComparer.Default);
```

Comment:
I'd use `ReadOnlyCollection` as value type of the dictionary, since these should never be modified.